accounting/account/models.py

```diff
@@ -0,0 +1,155 @@
+        return '{} {}, {} {} {}, {}'.format(
+            self.address_line1, self.address_line2, self.city, self.state, self.zipcode, self.country
+        )
+
+
+class Account(models.Model):
+    """
+    An account contains accounting details for a user, i.e. bank account details.
+
+    A user's accounts can have multiples of certain types of data.
+    For example, a user can have multiple bank accounts, multiple addresses, or multiple hourly rates
+    which differ per client where the user is the provider.
+    """
+
+    user = models.OneToOneField(
+        USER_MODEL, models.CASCADE, related_name='account',
+        help_text=_("The authentication user linked to this account."))
+    address = models.ForeignKey(
+        Address, models.CASCADE, related_name='accounts',
+        help_text=_("The address of this user."))
+    business_name = models.CharField(
+        max_length=120, blank=True, null=True,
+        help_text=_("The name of the business or company associated with this account."))
+    occupation = models.CharField(
+        max_length=80, blank=True, null=True,
+        help_text=_("The occupation of the user of this account. "
+                    "Required if country is US, CA, or JP."))
+
+    # We link hourly rates to other accounts, because an account can exist for both a provider and a client, and
+    # because hourly rates can differ for a provider per client, and vice versa. For example:
+    #     - Provider A has a X EUR/hour rate for Client F.
+    #     - Provider A has a Y EUR/hour rate for Client G.
+    #     - Client F pays X EUR/hour for Provider A.
+    #     - Client F pays Z EUR/hour for Provider B.
+    # For technical details, see
+    # https://docs.djangoproject.com/en/2.0/topics/db/models/#extra-fields-on-many-to-many-relationships
+    hourly_rate = models.ManyToManyField(
+        'self', through='HourlyRate', symmetrical=False,
+        help_text=_("The hourly rate the user charges clients."))
+
+    class Meta:
+        verbose_name = _('Account')
+        verbose_name_plural = _('Accounts')
+
+    def __str__(self):
+        """
+        Returns a string indicating to whom this account belongs.
+        """
+        return self.business_name or self.user.get_full_name() or self.user.username
+
+
+class HourlyRate(models.Model):
+    """
+    An hourly rate model that serves as a way to map a provider's hourly rate per client.
+
+    For technical details, see Django documentation, which explains intermediate models like this one:
+    https://docs.djangoproject.com/en/2.0/topics/db/models/#extra-fields-on-many-to-many-relationships
+    """
+
+    # Allows up to 999999.99 of an hourly rate for some currency.
+    # There are some currencies that are quite inflated. Or this provider is just really darn expensive!
+    hourly_rate = MoneyField(
+        max_digits=8, decimal_places=2,
+        help_text=_("The hourly rate charged between a provider and client."))
```

Comment:
Ah, I see, the hourly rates are generic, not just tied to our invoicing within the team, and the account models too. Not sure if we needed that level of abstraction though, since hourly rates only matter for the team itself - for others, we simply have incoming invoices, with a total, we don't care about the rate. It might have generated complexity in the models and the views needed to generate/edit them that was really necessary at this stage imho - a simple numerical field "hourly_rate" on a DeveloperAccount model would have been enough for now?

Reply:
Yeah this is quite generic, but it didn't complicate the views/serializers that much, actually. If you take a look at them, there's just one place that requires slight customization, otherwise it's the same as the others.

But I think this is a good starting point for handling outgoing invoices with hourly rates. Not entirely sure what I foresee but I don't think this hurts.

Reply:
OK, sounds good then, that seem reasonable.